packages/flutter/lib/src/cupertino/form_section.dart

```diff
@@ -0,0 +1,240 @@
+  /// lies above the [children] rows.
+  ///
+  /// The [children] parameter is required and sets the list of rows shown in
+  /// the section. The [children] parameter takes a list, as opposed to a more
+  /// efficient builder function that lazy builds, because forms are intended to
+  /// be short in row count. It is recommended that only [CupertinoFormRow] and
+  /// [CupertinoTextFormFieldRow] widgets be included in the [children] list in
+  /// order to retain the iOS look.
+  ///
+  /// The [margin] parameter sets the spacing around the content area of the
+  /// section encapsulating [children], and defaults to the standard
+  /// notched-style iOS form padding.
+  ///
+  /// The [borderRadius] parameter sets the circular border radius for the
+  /// [Column] encapsulating [children] rows. Defaults to 10.0 for the standard
+  /// edge-to-edge style.
+  ///
+  /// The [decoration] parameter sets the decoration for the section. Defaults
+  /// to a [CupertinoColors.secondarySystemBackground] background color.
+  const CupertinoFormSection.insetGrouped({
+    Key? key,
+    required this.children,
+    this.header,
+    this.margin = _kDefaultInsetGroupedRowsMargin,
+    this.borderRadius = _kDefaultInsetGroupedBorderRadius,
+    this.decoration,
+  })  : _type = _CupertinoFormSectionType.insetGrouped,
+        assert(children.length > 0),
+        super(key: key);
+
+  final _CupertinoFormSectionType _type;
+
+  /// Sets the form section header. The section header lies above the
+  /// [children] rows.
+  final Widget? header;
+
+  /// Margin around the content area of the section encapsulating [children].
+  ///
+  /// Defaults to zero padding if constructed with standard
+  /// [CupertinoFormSection] constructor. Defaults to the standard notched-style
+  /// iOS margin when constructing with [CupertinoFormSection.insetGrouped].
+  final EdgeInsetsGeometry margin;
+
+  /// The list of rows in the section.
+  ///
+  /// This takes a list, as opposed to a more efficient builder function that
+  /// lazy builds, because forms are intended to be short in row count. It is
+  /// recommended that only [CupertinoFormRow] and [CupertinoTextFormFieldRow]
+  /// widgets be included in the [children] list in order to retain the iOS look.
+  final List<Widget> children;
+
+  /// Sets the decoration for the section.
+  ///
+  /// Defaults to a [CupertinoColors.secondarySystemBackground] background
+  /// color.
+  final BoxDecoration? decoration;
+
+  /// Sets the border radius for the [Column] encapsulating [children]
+  /// rows.
+  ///
+  /// Defaults to zero radius when constructed with standard [CupertinoFormSection]
+  /// constructor. Defaults to 10.0 circular radius when constructing with
+  /// [CupertinoFormSection.insetGrouped].
+  final BorderRadius? borderRadius;
+
+  @override
+  Widget build(BuildContext context) {
+    final Color dividerColor = CupertinoColors.separator.resolveFrom(context);
+    const double dividerHeight = 0.5;
```

Comment:
perfect!

Reply:
Btw I did a change for this. Because the divider isn't a part of the row, there are color problems. In the inset-ed divider, the first 15 pixels are clear color which shows the background color of the form section (which is secondary background), but it should show the color of the rows (which is system background). What I did was place the inset-ed divider into a container that has a system background color. This works well as long as we don't open up the `decoration` attribute for `CupertinoFormRow`, because there will be a discrepancy in the colors.

What are your thoughts?

Reply:
How did we determine that the row was systemBackground and the section was secondarySystemBackground? Was it from the Xcode view hierarchy inspector? 

Intuitively it seems like they would be the same color but I might be missing something